### DISCUSSION TITLE
child_process.exec

### DISCUSSION THREAD

**Discussion by suntong** [deta/cloud-docs#560] [Help] [answered]
What's wrong with my following code:

```js
  const child_process = require("child_process")
  let hostname = 'NA'
  child_process.exec("hostname -f", function(err, stdout, stderr) {
    hostname = stdout.trim()
  })
```
I thought my hostname would now contain that from `hostname -f`, but it is still `'NA'`. What's wrong pls?

**Comment by lemonyte** [comment, marked as answer]
The `hostname` command doesn't exist on the Micro's OS.
![image](https://user-images.githubusercontent.com/49930425/198420162-3ffdf760-4227-4939-8e87-f7aebb7ed325.png)
